api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -631,7 +631,7 @@ def __len__(self) -> int:
         self._init()
         assert self._obs_joinids is not None
 
-        return len(self._obs_joinids)
+        return ceil(len(self._obs_joinids) / self.batch_size)
```

Comment:
Is there a case where you're going to get a floating point value from `len(self._obs_joinids) / self.batch_size` which has a remainder, even if the integer operation wouldn't?

I may do `divmod` here to be safe, but maybe this is fine. Not sure enough about floating point math here.

```suggestion
        div, rem = divmod(len(self._obs_joinids), self.batch_size)
        return div + bool(rem)
```